sql/sql_yacc.yy

```diff
@@ -15445,19 +15447,52 @@ alter_instance_stmt:
           ALTER INSTANCE_SYM alter_instance_action
           {
             Lex->sql_command= SQLCOM_ALTER_INSTANCE;
-            $$= NEW_PTN PT_alter_instance($3);
+            $$= NEW_PTN PT_alter_instance($3.alter_instance_action, $3.key_id);
           }
 
 alter_instance_action:
```

Comment:
Personally, I would reorganize the following grammar rule
```
alter_instance_action:
          ROTATE_SYM ident_or_text MASTER_SYM KEY_SYM
          | ROTATE_SYM ident_or_text SYSTEM_SYM KEY_SYM ulong_num
          | ROTATE_SYM ident_or_text SYSTEM_SYM KEY_SYM
          | RELOAD ident
          | RELOAD ident NO_SYM ROLLBACK_SYM ON_SYM ERROR_SYM
          ;
```
to something more unambiguous
```
alter_instance_action:
          ROTATE_SYM ident_or_text master_or_system_key_spec
          | RELOAD ident
          | RELOAD ident NO_SYM ROLLBACK_SYM ON_SYM ERROR_SYM
          ;
master_or_system_key_spec:
          MASTER_SYM KEY_SYM
          | SYSTEM_SYM KEY_SYM opt_key_id_spec
          ;
opt_key_id_spec:
          ulong_num
          |
          ;
```

Reply:
To be honest, I do not like it. You have to really read in - to know when there is system key rotation and when is master key rotation. You really do not see from this that system key can be only rotated from InnoDB. I think it just complicates simple think without strong reason to.

Reply:
OK, @robgolebiowski 
Your original solution is probably better from the "merging with upstream" point of view - it just adds a few more alternatives to an existing rule. So I am fine with it as long as it does not produce any new `bison` warnings.